iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -1626,6 +1630,12 @@ + (void)load {
     [OneSignalHelper registerAsUNNotificationCenterDelegate];
 }
 
++(BOOL) shouldDisableBasedOnProcessArguments {
+    if ([NSProcessInfo.processInfo.arguments containsObject: @"DISABLE_ONESIGNAL_SWIZZLING"]) {
```

Comment:
Is it possible for `processInfo` or `argument` to be `nil`? Possibly if you start the app from opening the app on the device instead of Xcode?

Reply:
This feature of disabling OneSignalSwizzling is dependent on injecting ProcessInfo variables. So it is reasonable that the `if condition` will become false when the processInfo.arguments is nil or empty.

Reply:
I wasn't sure on a real device when starting the app outside of Xcode if `processInfo` or `arguments` could be `nil` `containsObject` to throw. It seems like you have tested this but I'll give this a try as well to confirm as well.

Reply:
I tested this on iOS 14.4 and this didn't create any issues when starting the app outside of Xcode.